dashboard/app/controllers/code_review_comments_controller.rb

```diff
@@ -0,0 +1,83 @@
+class CodeReviewCommentsController < ApplicationController
+  load_and_authorize_resource
+  # TO DO: More nuanced permissioning. Sketched permissions below:
+  # create:
+  #   students (in same section as project owner?) and
+  #   teachers (only teacher leading section?) can create comments
+  # update (ie, change comment):
+  #   comment creator
+  #   teacher (only teacher leading section?)
+  # resolve:
+  #   owner of project (independent of account type) can update comments to resolved
+  # destroy:
+  #   comment creator
+  #   teacher (leading section?) can delete their own comments (or comments of their students)
+  # project_comments (get comments for a project):
+  #   project owner
+  #   teacher (leading section?)
+  #   any other student in section?
+
+  # POST /code_review_comments
+  def create
+    @code_review_comment = CodeReviewComment.new(code_review_comments_params)
+
+    if @code_review_comment.save
+      render json: @code_review_comment
+    else
+      head :bad_request
+    end
+  end
+
+  # TBD: will we update existing comments when someone wants to edit a comment,
+  # or just create a new comment?
+  # PATCH /code_review_comments/:id
+  def update
+    if @code_review_comment.update(code_review_comments_params)
+      head :ok
+    else
+      head :bad_request
+    end
+  end
+
+  # PATCH /code_review_comments/:id/resolve
+  def resolve
+    if @code_review_comment.update(is_resolved: true)
+      head :ok
+    else
+      head :bad_request
+    end
+  end
+
+  # DELETE /code_review_comments/:id
+  def destroy
+    if @code_review_comment.delete
+      head :ok
+    else
+      head :bad_request
+    end
+  end
+
+  # require project id, project version
+  # GET /code_review_comments/project_comments
+  def project_comments
+    @project_comments = CodeReviewComment.where(
+      channel_token_id: params[:channel_token_id],
+      project_version: params[:project_version]
+    )
+
+    render json: @project_comments
+  end
+
+  private
+
+  # TO DO: modify permit_params to handle other parameters (eg, section ID)
+  def code_review_comments_params
+    params.permit(
```

Comment:
this can be a follow-up, but I think the only piece of info the frontend has with regards to the channel is the encrypted channel id. We will either need to decrypt and store the storage_id and storage_app_id or decrypt and look up the channel_token_id based on the storage_id and storage_app_id. In commit code we are planning to just store the storage_id and storage_app_id instead of the channel_token_id. Sorry for the confusion (on my part) on this!

Reply:
Got it! Thinking I'll do this as an immediate follow-up to just keep things managable.